workers/unity/Assets/Generated/Source/improbable/gdk/tests/ExhaustiveBlittableSingularGameObjectComponentDispatcher.cs

```diff
@@ -39,26 +39,38 @@ internal class GameObjectComponentDispatcher : GameObjectComponentDispatcherBase
             {
             };
 
-            public override void InvokeOnAddComponentLifecycleCallbacks(GameObjectDispatcherSystem gameObjectDispatcherSystem)
+            protected override uint GetComponentId()
             {
-                var entities = ComponentAddedComponentGroup.GetEntityArray();
-                for (var i = 0; i < entities.Length; i++)
-                {
-                    var spatialOSBehaviourManager = gameObjectDispatcherSystem.GetSpatialOSBehaviourManager(entities[i].Index);
-                    spatialOSBehaviourManager.AddComponent(197720);
-                }
+                return 197720;
             }
 
-            public override void InvokeOnRemoveComponentLifecycleCallbacks(GameObjectDispatcherSystem gameObjectDispatcherSystem)
+            public override void InvokeOnComponentUpdateUserCallbacks(GameObjectDispatcherSystem gameObjectDispatcherSystem)
             {
-                var entities = ComponentRemovedComponentGroup.GetEntityArray();
+                var entities = ComponentsUpdatedComponentGroup.GetEntityArray();
+                var updateLists = ComponentsUpdatedComponentGroup.GetComponentArray<ComponentsUpdated<SpatialOSExhaustiveBlittableSingular.Update>>();
                 for (var i = 0; i < entities.Length; i++)
                 {
-                    var spatialOSBehaviourManager = gameObjectDispatcherSystem.GetSpatialOSBehaviourManager(entities[i].Index);
-                    spatialOSBehaviourManager.RemoveComponent(197720);
+                    var readers = gameObjectDispatcherSystem.GetSpatialOSBehaviourManager(entities[i].Index)
+                        .GetReadersWriters(197720);
```

Comment:
Why not call `GetComponentId` instead of embedding the literal?

Reply:
Could do that as well, I thought we might as well do the possibly marginally faster thing rather than a virtual function call when the ID is simply available to the codegen anyway.